freyja/sample_deconv.py

```diff
@@ -78,53 +69,35 @@ def map_to_constellation(vals,sample_strains,mapDict):
 				localDict['Other']=vals[jj]
 			else:
 				localDict['Other']+=vals[jj]
-
-	localDict = sorted(localDict.items(), key=lambda x: x[1], reverse=True)#convert to descending order
+	#convert to descending order
+	localDict = sorted(localDict.items(), key=lambda x: x[1], reverse=True)
 	return localDict
 
 
-def solve_demixing_problem(df_barcodes,df_mix,df_depths,mapDict):
-	sols = []
-	errors = []
-	strains = []
-	summarized = []
-	abundances = []
+def solve_demixing_problem(df_barcodes,mix,depths):
+	### single file problem setup, solving
 	eps = 1e-5#minimum abundance to include---TO DO:allow user to modify
-	for i in tqdm(range(df_mix.shape[0])):
-		dep= np.log2(df_depths.iloc[i]+1)
-		dep=dep/np.max(dep)#normalize depth scaling pre-optimization
-
-		# set up and solve demixing problem
-		A = np.array((df_barcodes*dep).T)
-		b = np.array(pd.to_numeric(df_mix.iloc[i])*dep)
-		x = cp.Variable(A.shape[1])
-		cost = cp.norm(A @ x - b,1)
-		constraints = [sum(x)==1,x>=0]
-		prob = cp.Problem(cp.Minimize(cost), constraints)
-		prob.solve(verbose=False)
-		sol = x.value
-		rnorm = cp.norm(A @ x - b, 1).value
-		#extract lineages with non-negligible abundance
-		sol[sol<eps] = 0
-		nzInds = np.nonzero(sol)[0]
-		newOrd = np.argsort(sol[nzInds])
-		sample_strains = df_barcodes.index[nzInds].to_list()
-		vals = sol[nzInds]/np.sum(sol[nzInds])
-		strains.append(sample_strains)
-
-		#### summarize by constellation
-		localDict = mapToConstellation(vals,sample_strains,mapDict)
-
-		abundances.append(sol[nzInds])
-		summarized.append(localDict)
-		sols.append(sol)
-		errors.append(rnorm)
-	sols_df = pd.DataFrame(data=np.stack((summarized,strains,abundances,errors),axis=1),index=[dmi.split('.')[0] for dmi in df_mix.index],columns=['summarized','lineages','abundances','resid'])
-	sols_df = sols_df.sort_index()
-	return sols_df
-
-
-
+	dep= np.log2(depths+1)
+	dep=dep/np.max(dep)#normalize depth scaling pre-optimization
+
+	# set up and solve demixing problem
+	A = np.array((df_barcodes*dep).T)
+	b = np.array(pd.to_numeric(mix)*dep)
+	x = cp.Variable(A.shape[1])
+	cost = cp.norm(A @ x - b,1)
+	constraints = [sum(x)==1,x>=0]
+	prob = cp.Problem(cp.Minimize(cost), constraints)
+	prob.solve(verbose=False)
+	sol = x.value
+	rnorm = cp.norm(A @ x - b, 1).value
+	#extract lineages with non-negligible abundance
+	sol[sol<eps] = 0
+	nzInds = np.nonzero(sol)[0]
+	newOrd = np.argsort(sol[nzInds])
+	sample_strains = df_barcodes.index[nzInds].to_list()
+	# vals = sol[nzInds]/np.sum(sol[nzInds])### that 
+	abundances = sol[nzInds]
+	return sample_strains,abundances,rnorm
 
 if __name__ == '__main__':
```

Comment:
This stuff below here would be pleasant to be done within the `_cli`

Reply:
💯